text/0000-partial-arithmetic.md

```diff
@@ -0,0 +1,54 @@
+
+There has been some discussions around Pony, mentioning that it is weird that division by zero is silently swallowed and transformed to `0`, while the rest of the language is focusing on safety. This has been considered kind of an inconsistency by quite a few people (from personal conversations). I know division has been deliberately changed to be non-partial, mostly for pragmatic reasons. This RFC wants to give the user a way to gain the safety back with a built-in operator, that is partial and errors on division by zero. This might not be too practical in day-to-day usage but it gives users the built-in tools to write safe arithmetic, if they desire to do so.
+
+For not having division stand out amongst the arithmetic operations, and for giving users an interface that is expectable ans consistent across operations, all other basic arithmetic operations should get a partial counterpart, that errors on underflow, overflow, comparable to the existing `addc`, `subc` and `mulc` functions.
+
+
+# Detailed design
+
+The operators need to be added in the parser. 
+
+They should be made customizable, so that users can write a partial `add` function for their objects, and make use of `+?` for adding two of their objects in a partial way. The method names should be the names for the base operations with a `_partial` suffix. E.g. `sub_partial` for partial subtraction.
+
+The actual implementation is quite simple, for addition, subtraction and multiplication we can use `addc`, `subc` and `mulc` and match on `overflow` flag and error if it is `true`. For division, we can simply check the divisor for being `0` and error in that case.
+
+# How We Teach This
+
+The new methods added need to have docstrings explaining their behavior. The docstrings of the other methods should mention the other method of doing e.g. addition.
+
+The tutorial should introduce the different ways of doing arithmetic in greater detail, *checked* (`addc` etc.), *unchecked* (`add` etc.). *unsafe* ( `add_unsafe` etc.) and *partial* (`add_partial`). So that users know what options they have and what the drawbacks might be (e.g. performance against correctness).
+
+# How We Test This
+
+How do we assure that the initial implementation works? How do we assure going forward that the new functionality works after people make changes? Do we need unit tests? Something more sophisticated? What's the scope of testing? Does this change impact the testing of other parts of Pony? Is our standard CI coverage sufficient to test this change? Is manual intervention required?
```

Comment:
I think this section of the RFC template still needs to be replaced with content?